pkg/shell/hosts_dialog.jsx

```diff
@@ -928,40 +1052,23 @@ export class HostModal extends React.Component {
 
     addressOrLabel() {
         const machine = this.props.machines_ins.lookup(this.state.address);
-        let host = this.props.machines_ins.split_connection_string(this.state.address).address;
+        let host = split_connection_string(this.state.address).address;
         if (machine && machine.label)
             host = machine.label;
         return host;
     }
 
-    changeContent(template, error_options) {
+    changeContent(template, error_options, with_error_message) {
         if (this.state.current_template !== template)
-            this.setState({ current_template: template, error_options });
+            this.setState({
+                current_template: template,
+                error_options,
+                dialogError: with_error_message ? cockpit.message(error_options) : null,
+            });
     }
 
     try2Connect(address, options) {
-        return new Promise((resolve, reject) => {
-            const conn_options = { ...options, payload: "echo", host: address };
-
-            conn_options["init-superuser"] = get_init_superuser_for_options(conn_options);
-
-            const machine = this.props.machines_ins.lookup(address);
-            if (machine && machine.host_key && !machine.on_disk) {
-                conn_options['temp-session'] = false; // Compatibility option
-                conn_options.session = 'shared';
-                conn_options['host-key'] = machine.host_key;
-            }
-
-            const client = cockpit.channel(conn_options);
-            client.send("x");
-            client.addEventListener("message", () => {
-                resolve();
-                client.close();
-            });
-            client.addEventListener("close", (event, options) => {
-                reject(options);
-            });
-        });
+        return try2Connect(this.props.machines_ins, address, options);
```

Comment:
This reads really strange at first, until I realized that `try2Connect` is both a global function *and* a method. Can you just eliminate the method (not sure how often it gets called, i.e. if adding `this.props.machines_ins` hurts much), or at least rename it?

Reply:
Yes, I'll clean this up. This is indeed just some minimum effort to plumb things together.

Reply:
Done.